CMakeLists.txt

```diff
@@ -26,59 +26,82 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
-cmake_minimum_required(VERSION 3.10.2)
-project(ot-efr32 VERSION 0.2.0)
 
+# ==============================================================================
+# CMake project definitions
+# ==============================================================================
+cmake_minimum_required(VERSION 3.16.0)
```

Comment:
Version increase is needed because we are using some newer CMake commands in the generated platform libraries